services/profile/controller/controller.go

```diff
@@ -72,6 +74,29 @@ func GetProfileById(w http.ResponseWriter, r *http.Request) {
 	json.NewEncoder(w).Encode(user_profile)
 }
 
+/*
+	GetProfileByUserId is used to get a profile for a provided user id.
+*/
+func GetProfileByUserId(w http.ResponseWriter, r *http.Request) {
+	user_id := mux.Vars(r)["id"]
+
+	profile_id, err := service.GetProfileIdFromUserId(user_id)
+
+	if err != nil {
+		errors.WriteError(w, r, errors.DatabaseError(err.Error(), "Could not get profile id associated with the user"))
```

Comment:
should we add "associated with the user **{user_id}**"? it gives better error information